### ISSUE TITLE
CLI

### ISSUE THREAD

**Issue by MartinThoma** [Storyyeller/Krakatau#156]
Currently, the scripts are harder to use than necessary. I think the situation would greatly improve by the following:

1. Adding a `setup.py`
2. Putting it on PyPI
3. Moving most of the functions in the `assemble.py`, `disassemble.py` and `decompile.py` to `Krakatau/cli` so that one can access the code from the package.
4. Creating a `bin/krakatau` CLI script which has `assmble`, `disassemble`, and `decompile` as subcommands

If you are interested in it, I could do it. This would add the dependency `click` (a commonly used CLI package) in the setup.py and in the `bin/krakatau`.

What do you think?

**Comment by Storyyeller** [comment]
In what was would this make Krakatau easier to use?

**Comment by MartinThoma** [comment]
Improvements by adding a setup.py, uploading to PyPI and having a single command line tool called like the package:

1. Discoverable via `pip search`
2. Tool can be installed via `pip install krakatau` - no need to know where the sources are
3. Functionality can be discovered by `krakatau --help`
4. No matter in which directory I am, I can simply execute `krakatau disassemble [foobar]` - no need to know where on my machine krakatau is.

**Comment by unoexperto** [comment]
I agree with @MartinThoma. Quick test is not working due to missing standard library. Path to what exactly (I see just bunch of `.so` files in `/usr/lib/jvm/java-11-oracle/lib`) I should pass to `-path` is unclear either. Documentation is almost nonexistent.

**Comment by unoexperto** [comment]
[This post](https://stackoverflow.com/a/27341150/226895) was very helpful. Perhaps it's worth including it to readme.txt

**Comment by Storyyeller** [comment]
I'm currently rewriting Krakatau in Rust, so the PyPI stuff is no longer relevant. But if you have any suggestions on how to best distribute/install it, I'm all ears.

**Comment by unoexperto** [comment]
@Storyyeller Robert, do you have repo with Rust-based version to share ?

**Comment by Storyyeller** [comment]
I haven't published the Rust version yet. I was planning to release an initial version for testing once the assembler was done.

**Comment by Storyyeller** [comment]
> do you have repo with Rust-based version to share ?

FYI, the Rust version is now live (it is on the "v2" branch). This only includes the assembler and disassembler for now - I'm still trying to figure out what to do for the new decompiler.

**Comment by Storyyeller** [comment]
Closing since this is obsolete with the move to Rust.
